### DISCUSSION TITLE
MQTT Client ID is always "watermeter"

### DISCUSSION THREAD

**Discussion by purehunter1** [jomjol/AI-on-the-edge-device#1145] [General] [answered]
Hi together,

I found, that the MQTT client ID is predefined as "watermeter" and cannot be changed in the Device configuration. (/jomjol_flowcontroll/ClassFlowMQTT.cpp).
In FHEM this leads to the problem, that all meters are called "watermeter". With the "main topic" configurable in the ESP the device, it is possible to distinguish between the devices, but basically they all have the Device name MQTT2_watermeter...

Am I wrong or can this get improved?

Beside this: really great project!!!

**Comment by caco3** [comment]
Did you set `MainTopic` in the MQTT settings (and tick the checkbox)? I guess its still on the default!

**Comment by purehunter1** [comment, marked as answer]
That is set correctly (in the settings menu).
But Main Topic is not the Client ID which I'm referring to.

But when looking into the config.ini I now saw that the Client ID can be set there. And now I also saw, that the client ID can be set in the settings menu also, in the Expert mode...

Ok. Solved.

rgds purehunter

**Reply by caco3** [reply]
Yes, the clientIId can be set separately.

BTW: No need to go to Expert mode for this!

But you are right, the default value "wasserzaehler" is not helpful.

**Reply by purehunter1** [reply]
You are right, Expert mode is not necessary. I just did not see/notice it, as the "greyed out" words for unchecked items are very light on my screen. I only saw it once I enabled it in the config.ini...
